### PR TITLE
dpv2 forecasting cli jobs

### PR DESCRIPTION
**Pull request by iamrk04**
# PR into Azure/azureml-examples

## Checklist

I have:

- [x] read and followed the contributing guidelines

## Changes

Add CLI v2 Example of:
- forecasting-energy-demand
- forecasting-forecast-function
- forecasting-orange-juice-sales
- forecasting-recipes-univariate

fixes #

### CODE REVIEW

File: cli/jobs/automl-standalone-jobs/cli-automl-forecasting-orange-juice-sales/test-mltable-folder/MLTable [hunk 1]
```diff
@@ -0,0 +1,6 @@
+paths:
+- file: ./oj-sales-test.csv
+transformations:
+- read_delimited:
+    delimiter: ','
+    encoding: ascii
```
Comment (vlbejan) [inline cli/jobs/automl-standalone-jobs/cli-automl-forecasting-orange-juice-sales/test-mltable-folder/MLTable:6]: I may be wrong, but I thought we need to use the convert_column_types parameter? This is what is being shown in the current v2 version (not created by our team) of the forecast [notebook](https://github.com/Azure/azureml-examples/blob/main/sdk/python/jobs/automl-standalone-jobs/automl-forecasting-task-energy-demand/data/test-mltable-folder/MLTable).

File: cli/jobs/automl-standalone-jobs/cli-automl-forecasting-orange-juice-sales/cli-automl-forecasting-orange-juice-sales.yml [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+  max_trials: 10
+  max_concurrent_trials: 4
+  trial_timeout_minutes: 15
+  enable_early_termination: true
+
+forecasting:
+  time_column_name: "WeekStarting"
+  forecast_horizon: 20
+  frequency: "W-THU"
+  target_lags: "auto"
```
Comment (vlbejan) [inline cli/jobs/automl-standalone-jobs/cli-automl-forecasting-orange-juice-sales/cli-automl-forecasting-orange-juice-sales.yml:34]: Target lags is not used in the SDK version. We should remove it to be consistent.

File: cli/jobs/automl-standalone-jobs/cli-automl-forecasting-orange-juice-sales/test-mltable-folder/MLTable (hunk 1 above)
Comment (iamrk04) [reply cli/jobs/automl-standalone-jobs/cli-automl-forecasting-orange-juice-sales/test-mltable-folder/MLTable:6]: @vlbejan
Nikolay confirmed that the convert_column_types is not required.

File: cli/jobs/automl-standalone-jobs/cli-automl-forecasting-orange-juice-sales/cli-automl-forecasting-orange-juice-sales.yml (hunk 2 above)
Comment (iamrk04) [reply cli/jobs/automl-standalone-jobs/cli-automl-forecasting-orange-juice-sales/cli-automl-forecasting-orange-juice-sales.yml]: removed

File: cli/jobs/automl-standalone-jobs/cli-automl-forecasting-recipes-univariate/cli-automl-forecasting-recipes-univariate.yml [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+  timeout_minutes: 20
+  max_trials: 10
+  max_concurrent_trials: 4
+  trial_timeout_minutes: 20
+  enable_early_termination: true
+
+forecasting:
+  time_column_name: "observation_date"
+  forecast_horizon: 12
+  frequency: "D"
```
Comment (vlbejan) [inline cli/jobs/automl-standalone-jobs/cli-automl-forecasting-recipes-univariate/cli-automl-forecasting-recipes-univariate.yml:33]: I believe the frequency is monthly.

**Review by vlbejan — APPROVED** [review]

File: cli/jobs/automl-standalone-jobs/cli-automl-forecasting-orange-juice-sales/cli-automl-forecasting-orange-juice-sales.yml [hunk 4]
```diff
@@ -0,0 +1,38 @@
+$schema: https://azuremlsdk2.blob.core.windows.net/preview/0.0.1/autoMLJob.schema.json
+type: automl
+
```
Comment (nick863) [inline cli/jobs/automl-standalone-jobs/cli-automl-forecasting-orange-juice-sales/cli-automl-forecasting-orange-juice-sales.yml:3]: We need to add the featurization configuration as it is done in V1 notebook

```
featurization:
  featurization_config:
    column_purposes:
      CPWVOL5: Numeric
    transformer_params:
      Imputer:
      - fields: ["Quantity"]
        parameters:
          strategy: constant
          fill_value: 0
      - fields: ["INCOME"]
        parameters:
          strategy: median
```

**Review by nick863 — COMMENTED** [review]
I have left one comment.

File: cli/jobs/automl-standalone-jobs/cli-automl-forecasting-recipes-univariate/cli-automl-forecasting-recipes-univariate.yml (hunk 3 above)
Comment (iamrk04) [reply cli/jobs/automl-standalone-jobs/cli-automl-forecasting-recipes-univariate/cli-automl-forecasting-recipes-univariate.yml]: Thanks for pointing this out. This is updated to "M".

File: cli/jobs/automl-standalone-jobs/cli-automl-forecasting-orange-juice-sales/cli-automl-forecasting-orange-juice-sales.yml (hunk 4 above)
Comment (iamrk04) [reply cli/jobs/automl-standalone-jobs/cli-automl-forecasting-orange-juice-sales/cli-automl-forecasting-orange-juice-sales.yml:3]: @nick863 
I have added this, also added for field "Price"

File: cli/jobs/automl-standalone-jobs/cli-automl-forecasting-recipes-univariate/cli-automl-forecasting-recipes-univariate.yml [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+  max_cores_per_trial: -1
+  trial_timeout_minutes: 5
+  enable_early_termination: true
+
+forecasting:
+  time_column_name: "observation_date"
+  time_series_id_column_names: []
+  use_stl: none
+  forecast_horizon: 12
+  frequency: "MS"
```
Comment (nick863) [inline cli/jobs/automl-standalone-jobs/cli-automl-forecasting-recipes-univariate/cli-automl-forecasting-recipes-univariate.yml:36]: let us remove frequency here for pairity with the notebook.

**Review by nick863 — APPROVED** [review]

File: cli/jobs/automl-standalone-jobs/cli-automl-forecasting-recipes-univariate/cli-automl-forecasting-recipes-univariate.yml (hunk 5 above)
Comment (iamrk04) [reply cli/jobs/automl-standalone-jobs/cli-automl-forecasting-recipes-univariate/cli-automl-forecasting-recipes-univariate.yml]: removed.

**Pull request merged**